internal/db/schema/migrations/oss/postgres/59/01_target_ingress_egress_worker_filters.up.sql

```diff
@@ -0,0 +1,137 @@
+begin;
+
+-- Update tables to add ingress_worker_filter and egress_worker_filter
+alter table session
+  add column egress_worker_filter wt_bexprfilter,
+  add column ingress_worker_filter wt_bexprfilter;
+
+alter table target_tcp
+  add column egress_worker_filter wt_bexprfilter,
+  add column ingress_worker_filter wt_bexprfilter;
+
+-- Trigger functions to ensure that worker_filter and ingress/egress_worker_filter are mutually exclusive
+-- and that worker_filter can only be updated
+create function validate_filter_values_on_insert() returns trigger
+as $$
+begin
+  if new.worker_filter is not null then
+    raise exception 'worker_filter is deprecated and cannot be set';
+  end if;
+
+  return new;
+
+end;
+$$ language plpgsql;
+
+create function validate_filter_values_on_update() returns trigger
+as $$
+begin
+  if new.egress_worker_filter is not null then
+    if new.worker_filter = old.worker_filter then
```

Comment:
What happens if these _aren't_ equivalent, e.g. if they do an update and both add ingress/egress fields but also update the existing worker filter? I think right now it will just accept that?

Reply:
If a user tries to update both, they'll get a warning that 'cannot set worker_filter and ingress_filter; they are mutually exclusive fields' and nothing will happen (this is checked lower in the trigger func). The old filter can only be cleared out if they *only* update egress/ ingress without updating the worker filter